hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/TcpIpJoiner.java

```diff
@@ -57,13 +59,29 @@
 
 public class TcpIpJoiner extends AbstractJoiner {
 
+    /**
+     * Specifies how long the address of a member that has previously joined the cluster
+     * will be retained in the knownMemberAddresses after the member leaves the cluster.
+     */
+    public static final String PREVIOUSLY_JOINED_MEMBER_ADDRESS_RETENTION_DURATION_PROP =
+            "hazelcast.previously.joined.member.address.retention.duration.seconds";
```

Comment:
I would suggest adding `internal` in the name. Eg `hazelcast.internal.previously.joined.member.address.retention.duration.seconds`.